ledger/ledger-common/src/main/kotlin/net/corda/v5/ledger/common/transactions/SignedTransaction.kt

```diff
@@ -0,0 +1,33 @@
+package net.corda.v5.ledger.common.transactions
+
+import net.corda.v5.application.crypto.DigitalSignatureAndMetadata
+import net.corda.v5.base.annotations.CordaSerializable
+import net.corda.v5.base.annotations.DoNotImplement
+import java.security.PublicKey
+
+@DoNotImplement
+@CordaSerializable
+interface SignedTransaction {
+    val wireTransaction: WireTransaction
+    val ledgerTransaction: LedgerTransaction //TODO(do we need this? Or rather delegate some properties through?)
+
+    val sigs: List<DigitalSignatureAndMetadata>
+
+    /** Specifies all the public keys that require signatures for the transaction to be valid. */
+    val requiredSigningKeys: List<PublicKey> //TODO(was a set previously.)
+
+    /** Returns the same transaction but with an additional (unchecked) signature. */
+    fun withAdditionalSignature(sig: DigitalSignatureAndMetadata): SignedTransaction
+
+    /** Returns the same transaction but with an additional (unchecked) signatures. */
+    fun withAdditionalSignatures(sigList: Iterable<DigitalSignatureAndMetadata>): SignedTransaction
+
+    /** Alias for [withAdditionalSignature] to let you use Kotlin operator overloading. */
+    operator fun plus(sig: DigitalSignatureAndMetadata) = withAdditionalSignature(sig)
```

Comment:
These are not Java API compatible. We might be able to do them as extension methods, but they should not be on the api interface

Reply:
Did you mean something like this?
https://github.com/corda/corda-api/pull/487/commits/b3c3c9e41a40f0c54fd9b1b74bf3a0d77a731c4e
Can this stay in corda-api? Or does it need to go to runtime-os?